src/commands/up.rs

```diff
@@ -134,13 +159,102 @@ impl UpCommand {
         })
     }
 
+    fn resolve_app_source(&self) -> AppSource {
+        match (
+            &self.app_source,
+            &self.file_source,
+            &self.bindle_source,
+            &self.registry_source,
+        ) {
+            (None, None, None, None) => self.default_manifest_or_none(),
+            (Some(source), None, None, None) => Self::infer_source(source),
+            (None, Some(file), None, None) => Self::infer_file_source(file.to_owned()),
+            (None, None, Some(id), None) => AppSource::Bindle(id.to_owned()),
+            (None, None, None, Some(reference)) => AppSource::OciRegistry(reference.to_owned()),
+            (_, _, _, _) => AppSource::unresolvable("More than one application was specified"),
```

Comment:
Entirely optional slight simplification:
```suggestion
            _ => AppSource::unresolvable("More than one application was specified"),
```

Reply:
Obvious in retrospect but TIL - thank you.